File: app/src/main/java/posidon/launcher/wall/WallAdapter.java
package posidon.launcher.wall;

import android.content.Context;
import android.graphics.drawable.ShapeDrawable;
import android.graphics.drawable.shapes.RectShape;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.palette.graphics.Palette;

import posidon.launcher.R;

class WallAdapter extends BaseAdapter {

    private final Context context;

    WallAdapter(Context context) { this.context = context; }

	@Override public int getCount() { return Gallery.walls.size(); }
    @Override public Object getItem(int position) { return null; }
    @Override public long getItemId(int position) { return 0; }

    static class ViewHolder {
        ImageView pic;
        TextView label;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final ViewHolder viewHolder;
        LayoutInflater li = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        if (convertView == null) {
            convertView = li.inflate(R.layout.wall_item, null);
            viewHolder = new ViewHolder();
            viewHolder.pic = convertView.findViewById(R.id.pic);
            viewHolder.label = convertView.findViewById(R.id.label);
            convertView.setTag(viewHolder);
        } else viewHolder = (ViewHolder)convertView.getTag();
        Wall wall = Gallery.walls.get(position);
        if (wall != null) {
            viewHolder.pic.setImageBitmap(wall.img);
            ShapeDrawable btnbg = new ShapeDrawable(new RectShape());
            try { btnbg.getPaint().setColor(0x00FFFFFF & Palette.from(wall.img).generate().getDarkMutedColor(context.getResources().getColor(R.color.walllabelbg)) | 0xaa000000); }
            catch (Exception e) { btnbg.getPaint().setColor(context.getResources().getColor(R.color.walllabelbg)); }
            viewHolder.label.setBackground(btnbg);
            viewHolder.label.setText(wall.name);
        }
        return convertView;
    }
}

File: app/src/main/java/posidon/launcher/wall/WallAdapter.kt
package posidon.launcher.wall

import android.content.Context
import android.graphics.drawable.ShapeDrawable
import android.graphics.drawable.shapes.RectShape
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.palette.graphics.Palette
import posidon.launcher.R

internal class WallAdapter(private val context: Context) : BaseAdapter() {

    override fun getCount(): Int = Gallery.walls.size
    override fun getItem(position: Int): Any? = null
    override fun getItemId(position: Int): Long = 0

    internal class ViewHolder(val pic: ImageView, val label: TextView)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        var convertView = convertView
        val viewHolder: ViewHolder
        val btnbg: ShapeDrawable
        if (convertView == null) {
            convertView = LayoutInflater.from(context).inflate(R.layout.wall_item, parent, false)
            viewHolder = ViewHolder(convertView.findViewById(R.id.pic), convertView.findViewById(R.id.label))
            convertView.tag = viewHolder
            btnbg = ShapeDrawable(RectShape())
        } else {
            viewHolder = convertView.tag as ViewHolder
            btnbg = viewHolder.label.background as ShapeDrawable
        }
        val wall: Wall = Gallery.walls[position]
        viewHolder.pic.setImageBitmap(wall.img)
        try { btnbg.paint.color = 0x00FFFFFF and Palette.from(wall.img!!).generate().getDarkMutedColor(context.resources.getColor(R.color.walllabelbg)) or -0x56000000 }
        catch (e: Exception) { btnbg.paint.color = context.resources.getColor(R.color.walllabelbg) }
        viewHolder.label.background = btnbg
        viewHolder.label.text = wall.name
        return convertView
    }
}